configuration/configuration.go

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package configuration
+
+import (
+	"errors"
+	"fmt"
+	"github.com/01org/ciao/payloads"
+	"gopkg.in/yaml.v2"
+	"net/url"
+)
+
+// we can have values set to default, except for
+//    scheduler { storage_uri }
+//    controller { compute_ca, compute_cert, identity_user, identity_password }
+//    launcher { compute_net, mgmt_net }
+//    image_service { url }
+//    identity_service { url }
+//
+// so we need to have at least those values set in our config
+//
+// TODO: proper validation of values set in yaml setup
+func validMinConf(conf *payloads.Configure) bool {
+	return (conf.Configure.Scheduler.ConfigStorageURI != "" &&
+		conf.Configure.Controller.HTTPSCACert != "" &&
+		conf.Configure.Controller.HTTPSKey != "" &&
+		conf.Configure.Controller.IdentityUser != "" &&
+		conf.Configure.Controller.IdentityPassword != "" &&
+		conf.Configure.Launcher.ComputeNetwork != "" &&
+		conf.Configure.Launcher.ManagementNetwork != "" &&
+		conf.Configure.ImageService.URL != "" &&
+		conf.Configure.IdentityService.URL != "")
+}
+
+func getPayload(conf *payloads.Configure) (payload []byte, err error) {
+	if !validMinConf(conf) {
+		return nil, errors.New(
+			"minimal configuration is not met or yaml is malformed")
+	}
+	payload, err = yaml.Marshal(&conf)
+	if err != nil {
+		return nil, err
+	}
+	return payload, nil
+}
+
+func fillDefaults(conf *payloads.Configure) {
+	conf.Configure.Scheduler.ConfigStorageType = payloads.Filesystem
+	conf.Configure.Controller.ComputePort = 8774
+	conf.Configure.ImageService.Type = payloads.Glance
+	conf.Configure.IdentityService.Type = payloads.Keystone
+	conf.Configure.Launcher.DiskLimit = true
+	conf.Configure.Launcher.MemoryLimit = true
+}
+
+// TODO: add etcd support related scheme(s)
+func discoverDriver(uriStr string) (storageType payloads.StorageType, err error) {
+	uri, err := url.Parse(uriStr)
+	if err != nil {
+		return storageType, err
+	}
+	if uri.Scheme != "file" {
+		return storageType, fmt.Errorf(
+			"Configuration URI Scheme '%s' not supported", uri.Scheme)
+	}
+	storageType = payloads.Filesystem
+	return storageType, nil
+}
+
+// ConfigurationPaylad fills a payloads.Configure struct
+// composed with the values from the configuration blob given
+func ConfigurationPayload(yamlConf []byte, conf *payloads.Configure) (err error) {
```

Comment:
Side note: `golint configuration/` will show you all the golint errors/warnings you have to fix. We require a clean golint output and enforce that through travis.